### PR TITLE
Convert markup generation to a function

### PR DESCRIPTION
**Pull request by eschnett**
Using a function to generate markup ensures that its temporary variables (e.g. `str`) are not visible any more later on, avoiding compiler warnings.

### CODE REVIEW

File: doc/make.jl [hunk 1]
```diff
@@ -46,18 +46,21 @@ end
...
+    splitted = split(str, "<!--- generated by $basename-update.jl: -->")
+    @assert length(splitted) == 2
+    replaced_links = replace(splitted[1], r"\[\#([0-9]*?)\]" => s"[#\g<1>](https://github.com/JuliaLang/julia/issues/\g<1>)")
+    write(
+        joinpath(@__DIR__, "src", "$basename.md"),
+        """
+        ```@meta
+        EditURL = "https://github.com/JuliaLang/julia/blob/master/$basename.md"
+        ```
+
```
Comment (inkydragon) [inline doc/make.jl:60]:
```suggestion

```

Comment (eschnett) [reply doc/make.jl:60]: Why should the empty line be removed?

My PR doesn't change the output, only how it is generated. Shouldn't a change to the generated output be a different PR?

Or am I misinterpreting your suggestion? Do you want to remove trailing white space only, keeping the line?

Comment (inkydragon) [reply doc/make.jl:60]: > Do you want to remove trailing white space only, keeping the line?

Yes, and my local tests show that these multi-line strings are also equivalent after deleting these spaces.

```julia
s0 =
    """
    ```@meta
    EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
    ```

    """
s1 =
        """
        ```@meta
        EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
        ```
    
        """
s2 =
        """
        ```@meta
        EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
        ```

        """
[s0, s1, s2]
# 3-element Vector{String}:
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n\n"
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n    \n"
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n\n"

s00 = 
    """
    ```@meta
    url=""
    ```

    """

s01 = 
    """
    ```@meta
    url=""
    ```
    
    """

s10 = 
        """
        ```@meta
        url=""
        ```

        """

s11 = 
        """
        ```@meta
        url=""
        ```
    
        """

s12 = 
        """
        ```@meta
        url=""
        ```
        
        """

[s00, s01, s10, s11, s12]
# 5-element Vector{String}:
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n    \n"
#  "```@meta\nurl=\"\"\n```\n\n"
```

**Review by vtjnash — COMMENTED** [review]
Do you just want a `let basename = "NEWS"` block here (since the goal seems to be avoiding introducing unnecessary new global names)?

**Comment by inkydragon** [comment]
@vtjnash 
I am going to [add HISTORY.md](https://github.com/JuliaLang/julia/issues/45034#issuecomment-1107826259) to the documentation as well. 
So defining a new function now looks acceptable.
I just need to add a new line of function calls: `generate_markdown("HISTORY")`

Can we merge this pr now?
cc: @ViralBShah

**Comment by ViralBShah** [comment]
@vtjnash Can we merge?

**Comment by ViralBShah** [comment]
Actually this is pretty straightforwad. Merging.

**Pull request merged**